tests/load_balancer_test.go

```diff
@@ -0,0 +1,280 @@
+			DocumentType:      reflect.TypeOf("foo"),
+			ExpectSuccess:     true,
+		},*/
+	}
+
+	var r driver.Response
+	// Setup context alternatives
+	contexts := []queryTestContext{
+		/*queryTestContext{driver.WithResponse(nil, &r), false},
+		queryTestContext{driver.WithResponse(context.Background(), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryCount(nil), &r), true},
+		queryTestContext{driver.WithResponse(driver.WithQueryCount(nil, true), &r), true},
+		queryTestContext{driver.WithResponse(driver.WithQueryCount(nil, false), &r), false},*/
+		queryTestContext{driver.WithResponse(driver.WithQueryBatchSize(nil, 1), &r), false},
+		/*queryTestContext{driver.WithResponse(driver.WithQueryCache(nil), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryCache(nil, true), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryCache(nil, false), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryMemoryLimit(nil, 60000), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryTTL(nil, time.Minute), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryBatchSize(driver.WithQueryCount(nil), 1), &r), true},
+		queryTestContext{driver.WithResponse(driver.WithQueryCache(driver.WithQueryCount(driver.WithQueryBatchSize(nil, 2))), &r), true},*/
+	}
+
+	// keep track of whether at least one request was forwarded internally to the
+	// correct coordinator behind the load balancer
+	someRequestForwarded := false
+
+	// Run tests for every context alternative
+	for _, qctx := range contexts {
+		ctx := qctx.Context
+		for i, test := range tests {
+			cursor, err := db.Query(ctx, test.Query, test.BindVars)
+			if err == nil {
+				// Close upon exit of the function
+				defer cursor.Close()
+			}
+			if test.ExpectSuccess {
+				if err != nil {
+					t.Errorf("Expected success in query %d (%s), got '%s'", i, test.Query, err)
+					continue
+				}
+				count := cursor.Count()
+				if qctx.ExpectCount {
+					if count != int64(len(test.ExpectedDocuments)) {
+						t.Errorf("Expected count of %d, got %d in query %d (%s)", len(test.ExpectedDocuments), count, i, test.Query)
+					}
+				} else {
+					if count != 0 {
+						t.Errorf("Expected count of 0, got %d in query %d (%s)", count, i, test.Query)
+					}
+				}
+				var result []interface{}
+				for {
+					hasMore := cursor.HasMore()
+					doc := reflect.New(test.DocumentType)
+					if _, err := cursor.ReadDocument(ctx, doc.Interface()); driver.IsNoMoreDocuments(err) {
+						if hasMore {
+							t.Error("HasMore returned true, but ReadDocument returns a IsNoMoreDocuments error")
+						}
+						break
+					} else if err != nil {
+						t.Errorf("Failed to result document %d: %s", len(result), err)
+					}
+					if !hasMore {
+						t.Error("HasMore returned false, but ReadDocument returns a document")
+					}
+					result = append(result, doc.Elem().Interface())
+					if (wasForwarded(&r)) {
```

Comment:
In go, if expressions do not need brackets. A `go fmt` will remove them automatically.